Python/backtracking/N-queens.py

```diff
@@ -0,0 +1,55 @@
+# You are given N, and for a given N x N chessboard, find a way to place N queens such that no queen can attack any other queen on the chess board. A queen can be killed when it lies in the same row, or same column, or the same diagonal of any of the other queens. You have to print all such configurations.
+
+# Sample input
+# 4
+
+def isSafeToPut(i,j,n,solution):
+
+    k=i-1
+    while k>=0:
+        if solution[k][j]:
+            return False
+        k-=1
+
+    row=i-1
+    col=j-1
+    while row>=0 and col>=0:
+        if solution[row][col]:
+            return False
+        row-=1
+        col-=1
+
+    row=i-1
+    col=j+1
+    while row>=0 and col<n:
+        if solution[row][col]:
+            return False
+        row-=1
+        col+=1
+
+    return True
+
+
+def nqueen(solution,n,i):
+    if i==n:
+        for i in range(n):
+            for j in range(n):
+                print(solution[i][j],end=" ")
+        print()
+        return False
+
+    for j in range(n):
+        solution[i][j]=1
```

Comment:
Can you try formatting your Code? If you are using an IDE like PyCharm, you can do that automatically via `Ctrl+Alt+L`